src/store/RootState.ts

```diff
@@ -1,4 +1,7 @@
 import QueryRepresentation from '@/sparql/QueryRepresentation';
 
-export interface RootState extends QueryRepresentation {
+export interface RootState {
```

Comment:
The reasoning for not doing this is #43
I'm torn here:
On the one hand, I would like, as you suggest, to have a pure representation of the domain logic in the state.
On the other hand, almost every bit of that logic comes with extra information that needs to also be represented. Like entity search results or errors for a specific component, etc. This becomes more complicate by considering whether the label of an entity is _really_ part of the query?

Merging those two feels wrong, having two trees with the same structure feels wrong as well.

Reply:
You definitely have a point, and yes, I agree that the Property's label probably shouldn't have made it into the `QueryRepresentation` type. How about leaving the two completely separate, and putting a `buildQueryRepresentationFromStoreState( state: RootState ): QueryRepresentation` step in between? There would be some redundancy, but mainly in the type declarations.

Off-topic: Interesting, I don't know if I agree with the search results making a round trip through the store. Couldn't the results stay local to a PropertyLookup/ItemLookup component that wraps `Lookup` and encapsulates the request logic?

Reply:
> You definitely have a point, and yes, I agree that the Property's label probably shouldn't have made it into the `QueryRepresentation` type. How about leaving the two completely separate, and putting a `buildQueryRepresentationFromStoreState( state: RootState ): QueryRepresentation` step in between? There would be some redundancy, but mainly in the type declarations.

Mh, yes that makes sense. Eventually, that will need to be a (lossy) two-way transformation between Store and QueryRepresentation.

From that QueryRepresentation we have then a one-way transformation into SPARQL and another two-way transformation into JSON(?) for encoding it in the URL.

I'll see if I can adjust the naming.

Reply:
> Off-topic: Interesting, I don't know if I agree with the search results making a round trip through the store. Couldn't the results stay local to a PropertyLookup/ItemLookup component that wraps `Lookup` and encapsulates the request logic?

Mh, I'm torn again.
- On the one hand it would make a lot of things easier to encapsulate all that into a single component and not having to worry about the action writing the search results into the correct place in the store and the getter getting it from the correct place.
- On the other hand it feels strange to have the logic of initiating the request in a vue component, i.e. the presentation layer, and not more in the business logic layer like an action.

:thinking: 

But wrapping that DS Lookup in a PropertyLookup component is certainly a good start. Maybe this will be easier to judge once that component is there.